onnxscript/tests/function_libs/torch_lib/extra_opinfo.py

```diff
@@ -446,6 +446,22 @@ def sample_inputs_col2im(op_info, device, dtype, requires_grad, **kwargs):
         yield opinfo_core.SampleInput(tensor, args=(output_size, kernel_size), kwargs=kwargs)
 
 
+def sample_inputs_index(op_info, device, dtype, requires_grad, **kwargs):
+    del op_info  # Unused
+    del kwargs  # Unused
+    make_arg = functools.partial(
+        torch_testing.make_tensor, dtype=dtype, device=device, requires_grad=requires_grad
+    )
+    s = 5
+    test_args = [
+        ([common_methods_invocations.index_variable(2, s, device=device)],),
+        # ([torch.tensor()],)
```

Comment:
@justinchuby will add more tests

Reply:
Sharing snippet I used to test. It poses complication here as indexing, i.e., `data[]`, lowers to multiple operators, usually couple of `aten.slice` + `aten.index` in the end. So you might need to figure out a way to construct the proper input state if you are testing `aten.index.Tensor` alone.

```python
import torch
import onnxruntime
import onnxscript


class IndexModel(torch.nn.Module):
    def forward(self, data, index, index2):
        # return data[..., index, index2]
        # return data[index, :, index2, ...]
        # return data[..., index, index2, :]
        # return data[index, :, index2]
        return data[:, :, index, index2]


data = torch.arange(0, 7 * 3 * 4 * 5 * 6).view(7, 3, 4, 5, 6)
index = torch.tensor([2, 1])
index2 = torch.tensor([[[0]], [[2]]])

model = IndexModel()
pt_output = model(data, index, index2)
print("PyTorch output shape:", pt_output.shape)

export_output = torch.onnx.dynamo_export(model, data, index, index2)
export_output.diagnostic_context.dump("log.sarif")

print(onnxscript.proto2text(export_output.model_proto))
sess = onnxruntime.InferenceSession(
    export_output.model_proto.SerializeToString(), providers=["CPUExecutionProvider"]
)
output = sess.run(None, {"arg0": data.numpy(), "arg1": index.numpy(), "arg2": index2.numpy()})

torch.testing.assert_close(pt_output.numpy(), output[0])

print("PASS")
```